src/activity_queue.rs

```diff
@@ -166,7 +166,8 @@ async fn sign_and_send(
         task.private_key.clone(),
         task.http_signature_compat,
     )
-    .await?;
+    .await
+    .context("signing request")?;
```

Comment:
Instead of these context it seems much cleaner if you add new variants to the error enum.

Reply:
I don't think it makes much sense to pass *all* errors as structured errors to the outside - e.g. in this case the private key is broken which is kinda something that has to be fixed manually anyways. But it's really useful to at least get a tiny bit of context for what happened, which is what these changes do, with just a tiny change instead of a large change